### PR TITLE
[Unix] Create a framework for certbot integration tests: PART 3d

### PR DESCRIPTION
**Pull request by adferrand**
Following #6821, this PR continues to convert certbot integration tests into certbot-ci.

This PR add tests covering checks on L397-429 in `tests/certbot-boulder-integration.sh`. Previous lines are covered with existing tests, or by #6946, #6947 and #6948.

### CODE REVIEW

File: certbot-ci/certbot_integration_tests/certbot_tests/test_main.py [hunk 1]
```diff
@@ -98,3 +98,48 @@ def test_renew_with_hook_scripts(context):
 
     assert_cert_count_for_lineage(context.config_dir, certname, 2)
     assert_hook_execution(context.hook_probe, 'deploy')
+
+
+def test_invalid_domain_with_dns_challenge(context):
+    """Test certificate issuance failure with DNS-01 challenge."""
+    # Manual dns auth hooks from misc are designed to fail if the domain contains 'fail-*'.
+    certs = ','.join([context.get_domain('dns1'), context.get_domain('fail-dns1')])
```
Comment (bmw) [inline certbot-ci/certbot_integration_tests/certbot_tests/test_main.py]: Can we change the name of this variable to names or domains?

Comment (adferrand) [reply certbot-ci/certbot_integration_tests/certbot_tests/test_main.py]: Done, to domains.

**Comment by bmw** [comment]
Thanks for fixing up the git problems!

**Comment by adferrand** [comment]
Your welcome ^^

I splitted the PR, it is normal that I take care of it consolidation.

**Pull request merged**